web/modules/mof/src/ModelViewBuilder.php

```diff
@@ -229,7 +232,84 @@ private function buildComponentList(int $class, array $evaluation, string $statu
 
     foreach ($components as $component) {
       $license = $evaluation[$class]['licenses'][$component->id] ?? null;
-      $build["{$status}_components"]['#items'][] = $component->name . ($license ? " [{$license}]" : '');
+
+      //Gets license and name URLs from the model array, null if not available.
+      $license_url = $model['license_data'][0]['licenses']['components'][$component->id]['license_path'] ?? null;
+      $name_url = $model['license_data'][0]['licenses']['components'][$component->id]['component_path'] ?? null;
+
+      // If the URL is invalid, set it to null.
+      try {
+        $license_url = Url::fromUri($license_url);
+      }
+      catch (\Exception $e) {
+        $license_url = null; 
+      }
+
+      // If the URL is invalid, set it to null.
+      try {
+        $name_url = Url::fromUri($name_url);
+      }
+      catch (\Exception $e) {
+        $name_url = null;
+      }
+
+     if ($license) {
+        //Both license and name URLs are available
+        if ($license_url and $name_url) {
+          $license_link = [
+            '#type' => 'link',
+            '#title' => $license,
+            '#url' => $license_url,
+            '#attributes' => ['target' => '_blank'],
```

Comment:
We should not force opening a new window. It's better to leave that to the user.
The same applies throughout.